File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        currzero = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[i], nums[currzero] = nums[currzero], nums[i]
                currzero += 1
        

File: 283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();
        int zero_index = 0;
        for (int i = 0; i < n; i++){
            if (nums[i] != 0){
                swap(nums[i], nums[zero_index]);
                zero_index++;
            }
        }
    }
};